src/server/command/worldedit/blueprint.zig

```diff
@@ -0,0 +1,170 @@
+		const storedBlueprint = clipboard.store(main.stackAllocator);
+		defer main.stackAllocator.free(storedBlueprint);
+
+		const fileName: []const u8 = ensureBlueprintExtension(main.stackAllocator, args[1]);
+		defer main.stackAllocator.free(fileName);
+
+		var blueprintsDir = try openDir("blueprints");
+		defer blueprintsDir.close();
+
+		try blueprintsDir.write(fileName, storedBlueprint);
+
+		std.log.info("Saved clipboard to blueprint file: {s}", .{fileName});
+		source.sendMessage("#00ff00Saved clipboard to blueprint file: {s}", .{fileName});
+	} else {
+		source.sendMessage("#ff0000Error: No clipboard content to save.", .{});
+	}
+}
+
+fn ensureBlueprintExtension(allocator: NeverFailingAllocator, fileName: []const u8) []const u8 {
+	if(!std.ascii.endsWithIgnoreCase(fileName, ".blp")) {
+		return std.fmt.allocPrint(allocator.allocator, "{s}.blp", .{fileName}) catch unreachable;
+	} else {
+		return allocator.dupe(u8, fileName);
+	}
+}
+
+fn blueprintDelete(args: []const []const u8, source: *User) !void {
+	if(args.len < 2) {
+		source.sendMessage("#ff0000**/blueprint delete** requires FILENAME argument.", .{});
+		return;
+	}
+	if(args.len >= 3) {
+		source.sendMessage("#ff0000Too many arguments for **/blueprint delete**. Expected 1 argument, FILENAME.", .{});
+		return;
+	}
+
+	const fileName: []const u8 = ensureBlueprintExtension(main.stackAllocator, args[1]);
+	defer main.stackAllocator.free(fileName);
+
+	var blueprintsDir = try openDir("blueprints");
+	defer blueprintsDir.close();
+
+	try blueprintsDir.dir.deleteFile(fileName);
+
+	std.log.info("Deleted blueprint file: {s}", .{fileName});
+	source.sendMessage("#ff0000Deleted blueprint file: {s}", .{fileName});
+}
+
+fn blueprintList(source: *User) !void {
+	var blueprintsDir = try std.fs.cwd().makeOpenPath("blueprints", .{.iterate = true});
+	defer blueprintsDir.close();
+
+	var directoryIterator = blueprintsDir.iterate();
+	var index: i32 = 0;
+
+	while(try directoryIterator.next()) |entry| {
+		if(entry.kind != .file) break;
+		if(!std.ascii.endsWithIgnoreCase(entry.name, ".blp")) break;
+
+		source.sendMessage("#ffffff{}#00ff00 {s}", .{index, entry.name});
+		index += 1;
+	}
+}
+
+fn blueprintLoad(args: []const []const u8, source: *User) !void {
+	if(args.len < 2) {
+		source.sendMessage("#ff0000**/blueprint load** requires FILENAME argument.", .{});
+		return;
+	}
+	if(args.len >= 3) {
+		source.sendMessage("#ff0000Too many arguments for **/blueprint load**. Expected 1 argument, FILENAME.", .{});
+		return;
+	}
+
+	const fileName: []const u8 = ensureBlueprintExtension(main.stackAllocator, args[1]);
+	defer main.stackAllocator.free(fileName);
+
+	var blueprintsDir = try openDir("blueprints");
+	defer blueprintsDir.close();
+
+	const storedBlueprint = try blueprintsDir.read(main.stackAllocator, fileName);
+	defer main.stackAllocator.free(storedBlueprint);
+
+	if(source.commandData.clipboard != null) {
+		source.commandData.clipboard.?.deinit(main.globalAllocator);
```

Comment:
Why not use a capture?
```zig
	if(source.commandData.clipboard) |oldClipboard| {
		oldClipboard.deinit(main.globalAllocator);
```

Reply:
`deinit` takes mutable pointer, I guess it doesn't have to.